rust/worker/src/execution/operators/metadata_filtering.rs

```diff
@@ -1331,35 +1135,624 @@ mod test {
+            record_segment.clone(),
+            metadata_segment.clone(),
+            data.clone(),
+            None,
+            None,
+            None,
+            None,
+        );
+
+        let res = operator
+            .run(&input)
+            .await
+            .expect("Error during running of operator");
+        assert_eq!(res.offset_ids, existing.clone().collect());
+
+        // A full scan within the user specified ids should yield matching records
+        let input = MetadataFilteringInput::new(
+            blockfile_provider.clone(),
+            record_segment.clone(),
+            metadata_segment.clone(),
+            data.clone(),
+            Some((31..=90).map(|i| format!("id_{}", i)).collect()),
+            None,
+            None,
+            None,
+        );
+
+        let res = operator
+            .run(&input)
+            .await
+            .expect("Error during running of operator");
+        assert_eq!(
+            res.offset_ids,
+            existing.clone().filter(|i| &31 <= i && i <= &90).collect()
+        );
+
+        // A $eq check on metadata should yield matching records
+        let where_clause = Where::DirectWhereComparison(DirectWhereComparison {
+            key: "mod_five".to_string(),
+            comp: WhereComparison::Primitive(PrimitiveOperator::Equal, MetadataValue::Int(2)),
+        });
+        let input = MetadataFilteringInput::new(
+            blockfile_provider.clone(),
+            record_segment.clone(),
+            metadata_segment.clone(),
+            data.clone(),
+            None,
+            Some(where_clause),
+            None,
+            None,
+        );
+
+        let res = operator
+            .run(&input)
+            .await
+            .expect("Error during running of operator");
+        assert_eq!(
+            res.offset_ids,
+            existing.clone().filter(|i| i % 5 == 2).collect()
+        );
+
+        // A $ne check on metadata should yield matching records
+        let where_clause = Where::DirectWhereComparison(DirectWhereComparison {
+            key: "even".to_string(),
+            comp: WhereComparison::Primitive(
+                PrimitiveOperator::NotEqual,
+                MetadataValue::Bool(false),
+            ),
+        });
+        let input = MetadataFilteringInput::new(
+            blockfile_provider.clone(),
+            record_segment.clone(),
+            metadata_segment.clone(),
+            data.clone(),
+            None,
+            Some(where_clause),
+            None,
+            None,
+        );
+
+        let res = operator
+            .run(&input)
+            .await
+            .expect("Error during running of operator");
+        assert_eq!(
+            res.offset_ids,
+            existing
+                .clone()
+                .filter(|i| i % 2 == 1 || i % 4 == 0)
+                .collect()
+        );
+
+        // A $lte check on metadata should yield matching records
```

Comment:
Can we break this test into pieces?  Each of these check comments should/could be a separate unit test.

Reply:
Will break it up in the future when we refactor the unit tests here. Currently the unit tests here have a lot of duplication to setup the test, which could be avoided with some shared utility functions.